### ISSUE TITLE
cancel context at the end of the scenario

### ISSUE THREAD

**Issue by draganm** [cucumber/godog#513]
### 🤔 What's the problem you're trying to solve?

When setting up mock servers running in separate go-routines, it would be useful when the context passed to the scenario `Before` hooks would be cancellable and be cancelled at the end of the scenario.

### ✨ What's your proposed solution?

Wrap the scenario context with `context.WithCancel()` before `Before` hooks are run and call the `cancel()` after last `After()` hook is being called.

### ⛏ Have you considered any alternatives or workarounds?

currently I achieve this with:

```go
func InitializeScenario(ctx *godog.ScenarioContext) {
	var cancel context.CancelFunc

	ctx.Before(func(ctx context.Context, sc *godog.Scenario) (context.Context, error) {
		ctx, cancel = context.WithCancel(ctx)
		return ctx, nil

	})

	ctx.After(func(ctx context.Context, sc *godog.Scenario, err error) (context.Context, error) {
		cancel()
		return ctx, nil
	})

}
```
it's not very elegant and gets repeated every time I use godog

### 📚 Any additional context?
My expectation was that this happens automatically, but only after trawling through the source code and running a couple of experiments it turned out it is not the case.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/ISSUE_TEMPLATE/feature_request.md)*

**Comment by vearutop** [comment]
Sounds like a reasonable idea, seems this can be implemented here https://github.com/cucumber/godog/blob/202882807b9e70ddc3f6def103edeb3313468958/suite.go#L424-L428 with deferred cancellation in the same function.

I'd be happy to review the PR. 👍

**Comment by vearutop** [comment]
Done in #514.
